src/software/new_geom/linear_spline2d.cpp

```diff
@@ -0,0 +1,103 @@
+#include "software/new_geom/linear_spline2d.h"
+
+#include <algorithm>
+#include <sstream>
+
+LinearSpline2d::LinearSpline2d(const std::vector<Point>& points) : knots(points)
+{
+    initLinearSegments(points);
+}
+
+LinearSpline2d::LinearSpline2d(const std::initializer_list<Point>& points) : knots(points)
+{
+    initLinearSegments(points);
+}
+
+const Point LinearSpline2d::getValueAt(double val) const
+{
+    if (val < 0.0 || val > 1.0)
+    {
+        std::stringstream ss;
+        ss << "Tried to evaluate spline at " << val
+           << ", which is outside of domain of the spline: [0,1]";
+        throw std::invalid_argument(ss.str());
```

Comment:
```
* @return value of spline evaluated at value val. If val not in [0,1]
*         then the closest endpoint
```

This function does not properly implement the spec of the interface. It should clamp `val` to the range `[0, 1]`

Reply:
After some thought, I think I prefer the implementation, changed the javadoc to match.